src/main/java/com/cen4010/bookstore/wishlist/WishListService.java

```diff
@@ -89,4 +72,14 @@ public void toCart(UUID bookId, UUID wishListId, UUID userId) {
     userRepository.save(user);
     wishListRepository.save(wishList);
   }
+
+  private void validateNewWishList(List<WishList> wishLists, String proposedName)
+      throws LimitExceededException {
+    if (wishLists.size() >= 3) {
+      throw new LimitExceededException("User cannot have more than 3 wishlists");
+    }
+    if (wishLists.stream().anyMatch(list -> list.getName().equals(proposedName))) {
+      throw new IllegalArgumentException("Wishlist of same type already made");
```

Comment:
Really nitpicky but probably should say "Wishlist of same *name* already made" ,otherwise looks good to me